files/SZWEEXEC/ZWEGEN00

```diff
@@ -0,0 +1,888 @@
+          if COMPARE(members.i.substitutions.n, sub) = 0 then do
+            call Print 'Substitution 'sub' already noted. Skipping.'
+            isFound = 1
+            leave
+          end
+        end
+        if isFound <> 1 then do
+          k = members.i.substitutions.0 + 1
+          members.i.substitutions.k = sub
+          members.i.substitutions.0 = k
+        end
+      end
+      if lastChar < 0 then do
+        leave
+      end
+      firstChar = POS('{', !contentToRead.j, lastChar) + 1
+    end
+  end
+end
+
+say 'All of the substitutions were found.'
+
+/*
+================================================================================
+    Invoke the edit macro on the substitutions for each member.
+================================================================================
+*/
+
+say 'Invoking the edit macro on each member.'
+
+do i = 1 to members.0
+    if members.i.substitutions.0 > 0 then do
+      do j = 1 to members.i.substitutions.0
+        d = jclCopy'('members.i.name')'
+        call Print 'Edit 'd'.'
+        old = '{'members.i.substitutions.j'}'
+        new = value('CFG.'members.i.substitutions.j)
+        queue old new
+        call Print 'Change' old 'to' new'.'
+        cmd = 'edit dataset('"'"d"'"') macro(zwechg)'
+        call Print cmd
+        address ispexec cmd
+        if RC <= 4 then do
+          call Print 'Edit successful.'
+        end
+        else do
+          say 'Stopping at 'd'.'
+          ExitWithRC(8)
+        end
+      end
+  end
+end
+
+say 'The edit macro was invoked on each member.'
+
+/*
+================================================================================
+    Add the job card to each member if filled out.
+================================================================================
+*/
+
+card.0 = 0
+
+do i = 0 to 99
+  if COMPARE(SYMBOL('CFG.setup.jobCard.'i), 'VAR') = 0 then do
+    j = card.0 + 1
+    card.j = VALUE('CFG.setup.jobCard.'i)
+    card.0 = j
+  end
+  else do
+    leave
+  end
+end
+
+if card.0 > 0 then do
+  say 'The job card has 'card.0' lines.'
+  say 'Adding the job card to each member.'
+  do i = 1 to members.0
+    if AllocateDataSet(jclCopy'('members.i.name')', 'zwejcl') > 0 then do
+      ExitWithRC(8)
+    end
+    if ReadFromDataSet('zwejcl') > 0 then do
+      ExitWithRC(8)
+    end
+    !contentToWrite.0 = !contentToRead.0 + card.0 - 1
+    do j = 1 to card.0
+      !contentToWrite.j = card.j
+    end
+    j = card.0 + 1
+    do k = 2 to !contentToRead.0
+      !contentToWrite.j = !contentToRead.k
+      j = j + 1
+    end
+    if WriteToDataSet('zwejcl') > 0 then do
+      ExitWithRC(8)
+    end
+    if FreeByDD('zwejcl') > 0 then do
+      ExitWithRC(8)
+    end
+  end
+  say 'The job card was added to each member.'
+end
+
+/*
+================================================================================
+  Validate(schema, yaml)
+================================================================================
+*/
+Validate:
```

Comment:
```
ZWEGENER.SYSTSPRT:
...
   390 +++  procedure expose !verbose                           
IRX0017I Error running ZWEGEN00, line 390: Unexpected PROCEDURE 
```

I believe, this is due to missing `exit` before the first procedure is defined. Without `exit`, REXX is just processing whatever is on the line.

Reply:
[Done.](https://github.com/zowe/zowe-install-packaging/pull/3718/commits/f59c01e37a84a73e331f7c66da26723d9e90860d)